File: src/test/java/com/technews/scheduler/SubscribeMailSchedulerTest.java
package com.technews.scheduler;

import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.ActiveProfiles;

@Disabled
@ActiveProfiles("local")
@SpringBootTest
class SubscribeMailSchedulerTest {

    @Autowired
    private SubscribeMailScheduler subscribeMailScheduler;

    @Test
    void run() {
        subscribeMailScheduler.runScheduler();
    }
}

File: src/test/java/com/technews/scheduler/SubscribeMailSchedulerTest.kt
package com.technews.scheduler

import com.technews.IntegrationTest
import com.technews.aggregate.subscribe.dto.SubscribeRequest
import com.technews.aggregate.subscribe.service.SubscribeService
import io.kotest.core.spec.style.StringSpec
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.transaction.annotation.Transactional

@Transactional
@IntegrationTest
class SubscribeMailSchedulerTest(
    @Autowired private val subscribeMailScheduler: SubscribeMailScheduler,
    @Autowired private val subscribeService: SubscribeService,
) : StringSpec({
        "subscribe mail".config(enabled = false) {
            val request = SubscribeRequest("jihunpark.tech@gmail.com")
            subscribeService.subscribe(request)
            subscribeMailScheduler.runSchedule()
        }
    })
